src/v2/Apps/Partner/Components/PartnerArtists/PartnerArtistsCarousel/PartnerArtistsCarousel.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Box,
+  Carousel,
+  CarouselCell,
+  CarouselRail,
+  Text,
+  Flex,
+  media,
+  Swiper,
+  themeProps,
+} from "@artsy/palette"
+import React from "react"
+import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
+import styled from "styled-components"
+import { useSystemContext } from "v2/Artsy"
+import { RouterLink } from "v2/Artsy/Router/RouterLink"
+import { useMatchMedia } from "v2/Utils/Hooks/useMatchMedia"
+import { Media } from "v2/Utils/Responsive"
+import { PartnerArtistsCarouselQuery } from "v2/__generated__/PartnerArtistsCarouselQuery.graphql"
+import { PartnerArtistsCarousel_partner } from "v2/__generated__/PartnerArtistsCarousel_partner.graphql"
+import {
+  PartnerArtistsCarouselItemFragmentContainer,
+  ResponsiveImage,
+} from "./PartnerArtistsCarouselItem"
+import { PartnerArtistsCarouselPlaceholder } from "./PartnerArtistsCarouselPlaceholder"
+
+const ArtistCarouselRail = styled(CarouselRail)`
+  li {
+    // 4 items per viewport
+    flex-basis: calc((100% - 3 * 20px) / 4);
+    flex-grow: 0;
+    flex-shrink: 0;
+    box-sizing: initial;
+  }
+
+  ${media.lg`
+    li {
+      // 3 items per viewport
+      flex-basis: calc((100% - 2 * 20px) / 3);
+    }
+  `};
+
+  ${media.md`
+    li {
+      // 2 items per viewport
+      flex-basis: calc((100% - 20px) / 2);
+    }
+  `};
+`
```

Comment:
If you find yourself doing this in any other place let me know and we'll come up with a re-useable way to do this. I don't love that we're essentially re-creating the grid behavior here.